scripts/end2end_examples.py

```diff
@@ -0,0 +1,119 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+To run this script you need to install papermill apart from argilla:
+
+$ pip install papermill
+
+Then run the script:
+
+$ python scripts/end2end_examples.py --help
+"""
+
+import os
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, Optional
+
+import papermill
+import typer
+
+EXAMPLES_FOLDER = (
+    Path(__file__).parent.parent / "docs/_source/tutorials_and_integrations/tutorials/feedback/end2end_examples"
+)
+OUTPUT_FOLDER = EXAMPLES_FOLDER / "output_notebooks"
+
+
+@dataclass
+class ExampleNotebook:
+    sort_index: int = field(init=False)
+    src_filename: Path
+    dst_filename: Path
+    parameters: Dict = field(default_factory=dict)
+
+    def __post_init__(self):
+        self.src_filename = EXAMPLES_FOLDER / self.src_filename
+        assert self.src_filename.exists(), f"File {self.src_filename} does not exist"
+        self.sort_index = int(self.src_filename.stem.split("-")[-1])
+        dst_folder = OUTPUT_FOLDER
+        if not dst_folder.exists():
+            dst_folder.mkdir(exist_ok=True)
+        self.dst_filename = dst_folder / self.dst_filename
+
+    def run(self):
+        try:
+            papermill.execute_notebook(
+                str(self.src_filename), str(self.dst_filename), parameters=self.parameters, kernel_name="python3"
+            )
+            print(f"✅  {self.src_filename.stem}")
+        except Exception as e:
+            print(f"❌  {self.src_filename.stem}")
+            raise e from None
+
+    def clean(self):
+        if self.dst_filename.exists():
+            self.dst_filename.unlink()
+            print(f"Removed output notebook: {self.dst_filename.stem}")
+
+
+def get_huggingface_token() -> Optional[str]:
+    if token := os.environ.get("HF_HUB_ACCESS_TOKEN"):
+        return token
+
+    from huggingface_hub import HfFolder
+
+    if token := HfFolder.get_token():
+        return token
+
+    raise ValueError("No token found. Please set HF_HUB_ACCESS_TOKEN environment variable.")
+
+
+def main(api_url: Optional[str] = None, api_key: Optional[str] = None, hf_token: Optional[str] = None) -> None:
```

Comment:
`api_url` and `api_key` are not used. I can imagine they must be used to setup the Argilla client for each tutorial, right?

Reply:
They were initially, but [this](https://github.com/argilla-io/argilla/pull/4208/files#diff-bbe8ef3e2a3d64e39e8d9dae60c31a354b1f47980ecab20a4113f73ac1bb2cd1R174-R179) addition from @davidberenstein1957 seems good enough. What do you think?

Reply:
If the environment override is not so difficult I think is better and more reusable to keep notebooks setup agnostic and let the runner configure the setup. 

So, I prefer to pass everything the notebook needs for the setup as parameters in the papermill call and let it set up the environment.